test_runner/regress/test_compatibility.py

```diff
@@ -195,6 +195,7 @@ def test_backward_compatibility(
 @check_ondisk_data_compatibility_if_enabled
 @pytest.mark.xdist_group("compatibility")
 @pytest.mark.order(after="test_create_snapshot")
+@pytest.mark.skip("Compression support breaks forward compatibility")
```

Comment:
We can't merge a change that breaks format compatibility: it makes it impossible to roll back the pageserver if something goes wrong -- that's what this test is for, to catch PRs that try to do that.

You need to think about your deployment plan: that probably means doing a release that can read the compressed format but never writes it, and only later enabling writes.

Reply:
Sorry, I do not completely understand your point.
Definitely this PR preserves backward compatibility. And backward compatibility test passed.
But seems to be obvious that old code, knowing nothing about this storage format change, can not work with data created by new version. So forward compatibility is not possible.

May regional suggestion was not to make any changes in storage format: just base on the assumption that we compress only 8kb pages and will not write compressed image kit its size is not smaller than original size. So we can just look at value size and if it is less than 8kb, then it is compressed. But @hlinnaka  didn't like this idea and prefer to bump storage format.

Concerning deployment: we can deploy page server with compression deployed by default. It will write layers in new format (having extra byte), but doesn't actually perform compression of data. Once this version is deployed, just rollback to any previous version is not possible any more, although nothing is compressed.  Then we enable compression for some tenants/regions/... and monitor it for some time (increased CPU usage, disk space saving,...).

Reply:
As I said:
> You need to think about your deployment plan: that probably means doing a release that can read the compressed format but never writes it, and only later enabling writes.

You don't have to preserve forward compatibilty forever, but we need at least one release cycle where the previous release can still read the new release's content.  I said above how to achieve that: you release version that can read the new format, then later a version that writes the new format.

Reply:
I can preserve old storage format version and write data in old format. But it will be able to read layers written in new format. The we can increment storage version and start writing storage in new format.